handlers/compute.go

```diff
@@ -1,45 +1,142 @@
 package handlers
 
 import (
+	"encoding/json"
 	"github.com/Sirupsen/logrus"
-	"github.com/rancher/agent/handlers/dockerClient"
+	"github.com/mitchellh/mapstructure"
+	"github.com/rancher/agent/handlers/docker"
 	"github.com/rancher/agent/handlers/progress"
 	"github.com/rancher/agent/handlers/utils"
 	"github.com/rancher/agent/model"
 	revents "github.com/rancher/go-machine-service/events"
 	"github.com/rancher/go-rancher/client"
-	"sync"
 )
 
-type InstanceWithLock struct {
-	mu sync.Mutex
-	in *model.Instance
-}
-
 func InstanceActivate(event *revents.Event, cli *client.RancherClient) error {
 	logrus.Infof("Received event: Name: %s, Event Id: %s, Resource Id: %s", event.Name, event.ID, event.ResourceID)
 	instance, host := utils.GetInstanceAndHost(event)
 
-	progress := progress.Progress{}
+	progress := progress.Progress{Request: event, Client: cli}
 
-	if instance != nil {
-		processData, ok := event.Data["processData"]
-		if ok {
-			instance.ProcessData = processData.(map[string]interface{})
-		}
+	if processData, ok := event.Data["processData"]; ok && instance != nil {
+		instance.ProcessData = processData.(map[string]interface{})
+	}
+	if utils.IsNoOp(event.Data) {
```

Comment:
You call IsNoOp both here and in utils.DoInstanceActivate. Is that just redundant or is there a legitimate need for it?